rust/cardano-blockchain-types/src/cip36/mod.rs

```diff
@@ -0,0 +1,130 @@
+//! CIP-36 Catalyst registration module
+
+pub mod key_registration;
+pub mod registration_witness;
+mod validation;
+pub mod voting_pk;
+
+use ed25519_dalek::VerifyingKey;
+use key_registration::Cip36KeyRegistration;
+use pallas::ledger::addresses::ShelleyAddress;
+use registration_witness::Cip36RegistrationWitness;
+use validation::{validate_payment_address_network, validate_signature, validate_voting_keys};
+use voting_pk::VotingPubKey;
+
+use crate::{MetadatumValue, Network};
+
+/// CIP-36 Catalyst registration
+#[derive(Clone, Default, Debug)]
+pub struct Cip36 {
+    /// Key registration - 61284
+    key_registration: Cip36KeyRegistration,
+    /// Registration witness - 61285
+    registration_witness: Cip36RegistrationWitness,
+    /// Is this a Catalyst strict registration?
+    is_catalyst_strict: bool,
+}
+
+/// Validation value for CIP-36.
+#[allow(clippy::struct_excessive_bools, clippy::module_name_repetitions)]
+#[derive(Clone, Default, Debug)]
+pub struct Cip36Validation {
+    /// Is the signature valid? (signature in 61285)
+    pub is_valid_signature: bool,
+    /// Is the payment address on the correct network?
+    pub is_valid_payment_address_network: bool,
+    /// Is the voting keys valid?
+    pub is_valid_voting_keys: bool,
+    /// Is the purpose valid? (Always 0 for Catalyst)
+    pub is_valid_purpose: bool,
+}
+
+impl Cip36 {
+    /// Get the `is_cip36` flag from the registration.
+    /// True if it is CIP-36 format, false if CIP-15 format.
+    #[must_use]
+    pub fn is_cip36(&self) -> Option<bool> {
+        self.key_registration.is_cip36
+    }
+
+    /// Get the voting public keys from the registration.
+    #[must_use]
+    pub fn voting_pks(&self) -> Vec<VotingPubKey> {
+        self.key_registration.voting_pks.clone()
```

Comment:
Perhaps it is better to return `&Vec<VotingPubKey>` here? In that case we can avoid clone if it isn't needed on the call sight and it is always possible to clone return value if it is needed.